Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -203,26 +213,44 @@ private void deriveTaskInfo(
     }
 
     private void deriveTimetable(
-        final ProcessSimulationInfo.ProcessSimulationInfoBuilder builder) {
-
-        builder.timetables(
-            Arrays.asList(Timetable.builder()
-                .defaultTimetable(true)
-                .id(config.getDefaultTimetable().get(CONFIG_DEFAULT_ID_KEY))
-                .name(config.getDefaultTimetable().get(CONFIG_DEFAULT_NAME_KEY))
-                .rules(Arrays.asList(Rule.builder()
-                    .id(UUID.randomUUID().toString())
-                    .name(config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_NAME_KEY))
-                    .fromWeekDay(DayOfWeek.valueOf(
-                        config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_FROM_WEEKDAY_KEY)
-                            .toUpperCase(DOCUMENT_LOCALE)))
-                    .toWeekDay(DayOfWeek.valueOf(
-                        config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_TO_WEEKDAY_KEY)
-                            .toUpperCase(DOCUMENT_LOCALE)))
-                    .fromTime(config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_FROM_TIME))
-                    .toTime(config.getDefaultTimetable().get(CONFIG_DEFAULT_TIMESLOT_TO_TIME))
-                    .build()))
-                .build()));
+        final ProcessSimulationInfo.ProcessSimulationInfoBuilder builder,
+        final CalendarModel calendarModel) {
+
+        if (calendarModel.getName().equals("Generic 24/7")) {
+            builder.timetables(List.of(createTimetable(
+                calendarModel, config.getTimetable().get(CONFIG_CUSTOM_ID_KEY), true)));
+        } else {
+            builder.timetables(
+                List.of(
+                    createTimetable(calendarModel, config.getTimetable().get(CONFIG_CUSTOM_ID_KEY), true),
+                    createTimetable(calendarService.getGenericCalendar(),
+                        config.getTimetable().get(CONFIG_DEFAULT_ID_KEY), false)
+                ));
```

Comment:
Can we enforce the rule there is only one default timetable in this list?

Reply:
The requirement was to also provide the 24/7 calendar as well if a custom calendar was specified.